lib/tapi_gtest/tapi_gtest.c

```diff
@@ -99,13 +71,19 @@ tapi_gtest_init(tapi_gtest *gtest, tapi_job_factory_t *factory)
         ),
     };
 
-    if ((rc = gtest_build_command(gtest, &gtest_args)) != 0)
+    if ((rc = te_string_append(&buf, "%s.%s", gtest->group, gtest->name)) != 0)
+    {
+        te_string_free(&buf);
+        return rc;
+    }
+    gtest->opts.gtest_filter = buf.ptr;
+    if ((rc = gtest_build_argv(gtest, &gtest_argv)) != 0)
         return rc;
```

Comment:
But now there is a memory leak if `gtest_build_argv` returns non-zero, the memory for `buf.ptr` is never freed.